src/main/java/spoon/support/DefaultCoreFactory.java

```diff
@@ -938,5 +968,45 @@ public CtTypeReference createWildcardStaticTypeMemberReference() {
 		result.setFactory(getMainFactory());
 		return result;
 	}
+	public CtModule createModule() {
+		CtModule module = new CtModuleImpl();
+		module.setFactory(getMainFactory());
+		return module;
```

Comment:
may be call `setParent(unnamedModule)` directly here? So it can be called directly by client?

Reply:
I don't agree with you on this point.

Reply:
I don't agree with you on this point: in my understanding, the default core factory is used to create model element, without given them any state except for the factory.

However, in order to put the module in the map, you need to have its name: you cannot have twice a module with the same name. 
So I vote to explicitely put in `ModuleFactory#getOrCreate` the mechanism to create a module and to put it in the map of modules properly. And to keep here, the creation of a simple CtModule detached from the model.